consensus/polybft/consensus_runtime.go

```diff
@@ -1144,7 +1134,26 @@ func (c *consensusRuntime) HasQuorum(
 			return c.fsm.validators.HasQuorum(signers)
 		}
 
-		return c.fsm.validators.HasQuorumWithoutProposer(signers)
+		if len(messages) == 0 {
+			return false
+		}
+
+		propAddress, exist := c.fsm.proposerCalculator.GetLatestProposer(messages[0].View.Round, blockNumber)
```

Comment:
This should be the job of the `proposerSnapshot`.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1018/commits/a46c6ec90b94f099b894f1e7aec34258d99b8b26